File: AtlasWeb/Core/Modules/Core_Destek_Modulleri/SMS_Module/SMSFormViewModel.cs
using System;
using System.Collections.Generic;
using TTObjectClasses;

namespace Core.Models
{
    public class SMSFormViewModel
    {
        public List<IstisnaiHal> IstisnaiHalDataSource { get; set; } = new List<IstisnaiHal>();
        public List<SKRSIlDTO> SKRSIlDataSource { get; set; } = new List<SKRSIlDTO>();
        public List<ProvizyonTipi> ProvizyonTipiDataSource { get; set; } = new List<ProvizyonTipi>();
        public List<SpecialityDefinitionDTO> SpecialityDefDataSource { get; set; } = new List<SpecialityDefinitionDTO>();
        public List<TedaviTuru> TedaviTuruDataSource { get; set; } = new List<TedaviTuru>();
    }

    #region Personel SMS Modeli
    public class SMSPersonnelFormViewModel
    {
        public List<SpecialityDefinition> SpecialityDataSource { get; set; } = new List<SpecialityDefinition>();
        public SMSPersonnelSearchModel PersonnelSearcModel { get; set; } = new SMSPersonnelSearchModel();
        public string SMSText;
    }

    public class SpecialityDefinitionDTO
    {
        public string Name { get; set; }
        public string Name_LowerCase { get; set; }
        public string Name_UpperCase { get; set; }
        public string Name_Shadow { get; set; }
        public Guid ObjectID { get; set; }
        public string Code { get; set; }

    }

    public class SMSPersonnelGridViewModel
    {
        //ResUser ObjectID
        public Guid ObjectID { get; set; }
        public string Name { get; set; }
        public string SurName { get; set; }
        public string Phone { get; set; }
        public string WorkStatus { get; set; }
        //ResUser daki Work alanı
        public string Position { get; set; }
    }

    public class SMSPersonnelSearchModel
    {
        public string FirstName { get; set; }
        public string SurName { get; set; }
        public int? WorkStatus { get; set; } = null;
        public int? Gender { get; set; } = null;
        public DateTime? BirthDate { get; set; } = null;
        public int? Title { get; set; } = null;
        public bool? HasPhoneNumber { get; set; } = true;
        public List<int> SelectedOccupations { get; set; } = new List<int>();
        //Doğum yeri şehir listesi
        public List<Guid> SelectedCitiesObjectIDs { get; set; } = new List<Guid>();
        //Görev yeri listesi
        public List<Guid> SelectedResourcesObjectIDs { get; set; } = new List<Guid>();
        public List<Guid> SelectedSpecialityObjectIDs { get; set; } = new List<Guid>();
    }
    #endregion Personel SMS Modeli

    #region Hasta SMS Modeli
    public class SMSPatientFormViewModel
    {
        public SMSPatientSearchModel PatientSearcModel { get; set; } = new SMSPatientSearchModel();
        public string SMSText;
    }

    public class SMSPatientGridViewModel
    {
        //Patient ObjectID
        public Guid ObjectID { get; set; }
        public string Name { get; set; }
        public string SurName { get; set; }
        public string Phone { get; set; }
        public string UniqueRefNo { get; set; }
    }

    public class SMSPatientSearchModel
    {
        public string FirstName { get; set; }
        public string SurName { get; set; }
        public int? Gender { get; set; } = null;
        public DateTime? BirthDate { get; set; } = null;
        public bool? HasPhoneNumber { get; set; } = true;
        public Guid? SelectedCityOfBirthPlace { get; set; }
        // Doğum yeri İlçesi
        public Guid? SelectedCountyOfBirthPlace { get; set; }
        //Adres İl
        public Guid? SelectedCityOfAddress { get; set; }
        //Adres İlçe
        public Guid? SelectedCountyOfAddress { get; set; }
        //Uyruk
        public Guid? Nationality { get; set; }
        //Kurum
        public List<Guid> SelectedPayers { get; set; }
        //Geliş sebebi
        public List<Guid> SelectedProvizyonTipi { get; set; }
        //İstisnai Hal
        public List<Guid> SelectedIstisnaiHal { get; set; }
        //Tanı
        public List<Guid> SelectedDiagnosis { get; set; }
        //Tedavi Türü
        public Guid? SelectedTreatment { get; set; }
        //Kabul tarihi başlangıç
        public DateTime? AdmissionStartDate { get; set; }
        //Kabul tarihi bitiş
        public DateTime? AdmissionEndDate { get; set; }
        //Yatış tarihi başlangıç
        public DateTime? InPatientStartDate { get; set; }
        //Yatış tarihi başlangıç
        public DateTime? InPatientEndDate { get; set; }
        public List<Guid> SelectedAdmissionDoctor { get; set; }
        public List<Guid> SelectedSpecialities { get; set; }
        public List<Guid> SelectedPoliclinics { get; set; }
    }
    #endregion Hasta SMS Modeli

    public class ResourceDTO
    {
        public string Name { get; set; }
        public string Name_Shadow { get; set; }
        public Guid ObjectId { get; set; }
    }

    public class SKRSIlDTO
    {
        public string ADI_LowerCase { get; set; }
        public string ADI_UpperCase { get; set; }

        public int? KODU { get; set; }
        public Guid ObjectID { get; set; }
        public string ADI { get; set; }
    }

    public class DiagnosisDefinitionDTO
    {
        public string Code { get; set; }
        public Guid ObjectID { get; set; }
        public string Name { get; set; }
        public string Name_Shadow { get; set; }
    }

    public class SKRSIlceDTO
    {
        public int? ILKODU { get; set; }
        public int? KODU { get; set; }
        public Guid? ObjectID { get; set; }
        public string ADI { get; set; }
    }

    public class SendPersonnelSMSModel
    {
        public List<SMSPersonnelGridViewModel> SMSModel { get; set; }
        public string SMSText { get; set; }
        public SMSTypeEnum? SMSType { get; set; }
    }

    public class SendPatientSMSModel
    {
        public List<SMSPatientGridViewModel> SMSModel { get; set; }
        public string SMSText { get; set; }
        public SMSTypeEnum? smsType { get; set; }
    }

    public enum SendSMSResultEnum
    {
        Successful = 0,
        UnSuccessful = 1,
        Cancelled = 2
    }
}

File: AtlasWeb/Core/Modules/Core_Destek_Modulleri/SMS_Module/SMSFormViewModel.ts
import { listboxObject } from "app/Invoice/InvoiceHelperModel";
import { EnumItem } from "app/NebulaClient/Mscorlib/EnumItem";
import { IEnumList } from "app/NebulaClient/Mscorlib/IEnumList";
import { ClassType } from "app/NebulaClient/ClassTransformer";
import { Guid } from "app/NebulaClient/Mscorlib/Guid";
import { City, SpecialityDefinition } from "app/NebulaClient/Model/AtlasClientModel";


export class SMSFormViewModel {
    public SKRSIlDataSource: Array<any> = new Array<any>();
    public ProvizyonTipiDataSource: Array<any> = new Array<any>();
    public IstisnaiHalDataSource: Array<any> = new Array<any>();
    public TedaviTuruDataSource: Array<any> = new Array<any>();
    public SpecialityDefDataSource: Array<SpecialityDefinition> = new Array<SpecialityDefinition>();
}

//#region SMS Personel Modeli
export class SMSPersonnelSearchModel {
    public FirstName: string;
    public SurName: string;
    public WorkStatus?: number = null;
    public Gender?: number = null;
    public BirthDate?: Date = null;
    public Title?: number = null;
    public HasPhoneNumber?: boolean = true;
    public SelectedOccupations?: Array<number> = new Array<number>();
    public SelectedCitiesObjectIDs: Array<Guid> = new Array<Guid>();
    public SelectedResourcesObjectIDs: Array<Guid> = new Array<Guid>();
    public SelectedSpecialityObjectIDs: Array<Guid> = new Array<Guid>();
}

export class SMSPersonnelFormViewModel {
    public PersonnelSearcModel: SMSPersonnelSearchModel = new SMSPersonnelSearchModel();
    public GridViewModel: Array<SMSPersonnelGridViewModel> = new Array<SMSPersonnelGridViewModel>();
    public SMSText: string;
}

export class SMSPersonnelGridViewModel {
    public ObjectID: Guid;
    public Name: string;
    public SurName: string;
    public Phone: string;
    //Çalışma Durumu
    public WorkStatus: string;
    //ResUser daki Work alanı (Görevi)
    public Position: string;
}
//#endregion


//#region Hasta SMS Modeli
export class SMSPatientSearchModel {
    public FirstName: string;
    public SurName: string;
    public Gender?: number = null;
    public BirthDate?: Date = null;
    public HasPhoneNumber?: boolean = true;
    //Doğum yeri şehir
    public SelectedCityOfBirthPlace?: Guid;
    // Doğum yeri İlçesi
    public SelectedCountyOfBirthPlace?: Guid;
    //Adres İl
    public SelectedCityOfAddress?: Guid;
    //Adres İlçe
    public SelectedCountyOfAddress?: Guid;
    //Uyruk
    public Nationality?: Guid;
    //Kurum
    public SelectedPayers: Array<Guid>;
    //Geliş sebebi
    public SelectedProvizyonTipi: Array<Guid>;
    //İstisnai Hal
    public SelectedIstisnaiHal: Array<Guid>;
    //Tanı
    public SelectedDiagnosis: Array<Guid>;
    //Tedavi Türü
    public SelectedTreatment: Guid;
    //Kabul tarihi başlangıç
    public AdmissionStartDate?: Date;
    //Kabul tarihi bitiş
    public AdmissionEndDate?: Date;
    //Yatış tarihi başlangıç
    public InPatientStartDate?: Date;
    //Yatış tarihi başlangıç
    public InPatientEndDate?: Date;
    //Kabul Doktoru
    public SelectedAdmissionDoctor: Array<Guid>;
    //Kabul Branşı
    public SelectedSpecialities: Array<any>;
    //Kabul Birimi
    public SelectedPoliclinics: Array<any>;

    public SetNullAdmissionStartDateRelatedCriterias() {
        this.SelectedPayers = new Array<Guid>();
        this.SelectedProvizyonTipi = new Array<Guid>();
        this.SelectedIstisnaiHal = new Array<Guid>();
        this.SelectedDiagnosis = new Array<Guid>();
        this.SelectedTreatment = null;
        this.AdmissionEndDate = null;
        this.SelectedAdmissionDoctor = new Array<any>();
        this.SelectedSpecialities = new Array<any>();
        this.SelectedPoliclinics = new Array<any>();
    }
}

//#region SMS Log Sorgulama
export class SMSLogSearchModel {
    public StartDate?: Date;
    public EndDate?: Date;
    public PhoneNumber: string;
    public Status?: boolean;
    public MessageText: string;
    public SMSType:number;
}

//#endregion

export class SMSPatientFormViewModel {
    public PatientSearchModel: SMSPatientSearchModel = new SMSPatientSearchModel();
    public GridViewModel: Array<SMSPatientGridViewModel> = new Array<SMSPatientGridViewModel>();
    public SMSText: string;
}

export class SMSPatientGridViewModel {
    public ObjectID: Guid;
    public Name: string;
    public SurName: string;
    public Phone: string;
    public UniqueRefNo: string;
}
//#endregion  Hasta SMS Modeli

export class SendPersonnelSMSModel {
    public SMSModel: Array<SMSPersonnelGridViewModel> = new Array<SMSPersonnelGridViewModel>();
    public SMSText: string;
    public SMSType: number;
}

export class SendPatientSMSModel {
    public SMSModel: Array<SMSPatientGridViewModel> = new Array<SMSPatientGridViewModel>();
    public SMSText: string;
    public SMSType: number;
}

export enum WorkStatusEnum {
    NotWorking = 0,
    Working = 1
}

export namespace WorkStatusEnum {
    export const _NotWorking = new EnumItem(WorkStatusEnum.NotWorking, 'Çalışmıyor', i18n('M13837', 'Çalışmıyor'), <number>WorkStatusEnum.NotWorking);
    export const _Working = new EnumItem(WorkStatusEnum.Working, 'Çalışıyor', i18n('M17061', 'Çalışıyor'), <number>WorkStatusEnum.Working);
    export const Items: Array<EnumItem> = [_NotWorking, _Working];

    @ClassType()
    export class WorkStatusEnumList implements IEnumList {
        get Items(): Array<EnumItem> {
            return WorkStatusEnum.Items;
        }
    }
}
